### PR TITLE
[WIP] add warning to `KLUFactorization`

### PR DESCRIPTION
**Pull request by vpuri3**

### CODE REVIEW

**Comment by vpuri3** [comment]
@Wimmerer

File: src/factorization.jl [hunk 1]
```diff
@@ -308,7 +308,13 @@ end
 function init_cacheval(alg::KLUFactorization, A, b, u, Pl, Pr, maxiters::Int, abstol,
                        reltol,
                        verbose::Bool, assumptions::OperatorAssumptions)
-    return KLU.KLUFactorization(convert(AbstractMatrix, A)) # this takes care of the copy internally.
+    M = if issparse(A)
+        A
+    else
+        @warn "nonsparse operator pased to KLUFactorization(). converting to sparse"
```
Comment (ChrisRackauckas) [inline src/factorization.jl:314]: It should just error

**Review by ChrisRackauckas — CHANGES_REQUESTED** [review]

File: src/factorization.jl (hunk 1 above)
Comment (Wimmerer) [reply src/factorization.jl:314]: I agree, I'm surprised it doesn't already in KLU itself?

Comment (vpuri3) [reply src/factorization.jl:314]: it does. but that's just breaking tests here: https://github.com/SciML/SciMLBase.jl/pull/210.

https://github.com/SciML/SciMLBase.jl/actions/runs/3109373156/jobs/5039530974

This PR is the quickest fix

Comment (vpuri3) [reply src/factorization.jl:314]: This PR does warn the user if a dense arrays is being passed, and is a no-op otherwise. So this should be harmless

Comment (ChrisRackauckas) [reply src/factorization.jl:314]: > but that's just breaking tests here

No, that test is doing something wrong. This is not a fix.

**Pull request closed without merging**